app/src/main/java/io/appium/uiautomator2/utils/ScreenshotHelper.java

```diff
@@ -147,17 +158,51 @@ public static String takeScreenshot() throws TakeScreenshotException {
         }
     }
 
+    public static byte[] compressJpeg(final Bitmap bitmap, float scale, int quality, boolean filter) throws TakeScreenshotException {
+        Bitmap resultBitmap;
+        if (scale == 1.0) {
```

Comment:
we cannot compare float number for equality. Check https://stackoverflow.com/questions/1088216/whats-wrong-with-using-to-compare-floats-in-java

Reply:
It seems like overkill, but I've taken the suggestion from the referenced link and this will be resolved w/ the upcoming update(s)

Reply:
image scaling hits performance. This means that the scaling will always be performed does not matter which scale factor we provide with the current implementation.

Reply:
That's correct, to your point, I'd like to avoid unnecessary CPU hits if at all possible.

Reply:
Understood.